src/ui/control/geolocate_control.js

```diff
@@ -423,8 +436,18 @@ class GeolocateControl extends Evented {
                 this._geolocateButton.classList.add('mapboxgl-ctrl-geolocate-waiting');
                 this._geolocateButton.setAttribute('aria-pressed', 'true');
 
+                numberOfWatches++;
+                let options;
```

Comment:
Just a personal preference, but I'd name this positionOptions so it's a bit clearer it's a https://developer.mozilla.org/en-US/docs/Web/API/PositionOptions object.